integrations-and-supported-tools/mlflow/notebooks/Neptune_MLflow.ipynb

```diff
@@ -0,0 +1,558 @@
+    "\n",
+    "import logging\n",
+    "import sys\n",
+    "import warnings\n",
+    "from urllib.parse import urlparse\n",
+    "\n",
+    "import mlflow.sklearn\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "from mlflow.models.signature import infer_signature\n",
+    "from sklearn.linear_model import ElasticNet\n",
+    "from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "\n",
+    "logging.basicConfig(level=logging.WARN)\n",
+    "logger = logging.getLogger(__name__)\n",
+    "\n",
+    "\n",
+    "def eval_metrics(actual, pred):\n",
+    "    rmse = np.sqrt(mean_squared_error(actual, pred))\n",
+    "    mae = mean_absolute_error(actual, pred)\n",
+    "    r2 = r2_score(actual, pred)\n",
+    "    return rmse, mae, r2\n",
+    "\n",
+    "\n",
+    "if __name__ == \"__main__\":\n",
+    "    warnings.filterwarnings(\"ignore\")\n",
+    "    np.random.seed(40)\n",
+    "\n",
+    "    # Read the wine-quality csv file from the URL\n",
+    "    csv_url = (\n",
+    "        \"https://raw.githubusercontent.com/mlflow/mlflow/master/tests/datasets/winequality-red.csv\"\n",
+    "    )\n",
+    "    try:\n",
+    "        data = pd.read_csv(csv_url, sep=\";\")\n",
+    "    except Exception as e:\n",
+    "        raise Exception(f\"Unable to download training data. Error: {e}\")\n",
+    "\n",
+    "    # Split the data into training and test sets. (0.75, 0.25) split.\n",
+    "    train, test = train_test_split(data)\n",
+    "\n",
+    "    # The predicted column is \"quality\" which is a scalar from [3, 9]\n",
+    "    train_x = train.drop([\"quality\"], axis=1)\n",
+    "    test_x = test.drop([\"quality\"], axis=1)\n",
+    "    train_y = train[[\"quality\"]]\n",
+    "    test_y = test[[\"quality\"]]\n",
+    "\n",
+    "    alpha = 0.5\n",
+    "    l1_ratio = 0.5\n",
+    "\n",
+    "    with mlflow.start_run():\n",
+    "        lr = ElasticNet(alpha=alpha, l1_ratio=l1_ratio, random_state=42)\n",
+    "        lr.fit(train_x, train_y)\n",
+    "\n",
+    "        predicted_qualities = lr.predict(test_x)\n",
+    "\n",
+    "        (rmse, mae, r2) = eval_metrics(test_y, predicted_qualities)\n",
+    "\n",
+    "        print(\"Elasticnet model (alpha={:f}, l1_ratio={:f}):\".format(alpha, l1_ratio))\n",
+    "        print(f\"  RMSE: {rmse}\")\n",
+    "        print(f\"  MAE: {mae}\")\n",
+    "        print(f\"  R2: {r2}\")\n",
+    "\n",
+    "        mlflow.log_param(\"alpha\", alpha)\n",
+    "        mlflow.log_param(\"l1_ratio\", l1_ratio)\n",
+    "        mlflow.log_metric(\"rmse\", rmse)\n",
+    "        mlflow.log_metric(\"r2\", r2)\n",
+    "        mlflow.log_metric(\"mae\", mae)\n",
+    "\n",
+    "        predictions = lr.predict(train_x)\n",
+    "        signature = infer_signature(train_x, predictions)\n",
+    "\n",
+    "        # Model registry does not work with Neptune URI\n",
```

Comment:
Is there enough "explanatory power" in this comment? Just wondering if it could be clarified a bit here and in the script.